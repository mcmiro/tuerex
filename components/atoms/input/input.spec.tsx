/* eslint-disable no-use-before-define */
import * as React from 'react';
import { screen, render, fireEvent } from '@testing-library/react';
import Input from './';

describe('Test input component', () => {
  it('Should pass when name, variant, size and placeholder is additional', () => {
    render(
      <Input
        name="test"
        variant="outline"
        size="md"
        placeholder="test-placeholder"
      />
    );

    const renderInputComponent = screen.getByRole('textbox');
    expect(renderInputComponent).toBeDefined();
    expect(renderInputComponent).toMatchSnapshot();
  });

  it('Should pass when value is changed', () => {
    const changeMock = jest.fn();
    const value = '';
    render(
      <Input
        name="test"
        value={value}
        onInputChange={changeMock}
        variant="outline"
        size="md"
        placeholder="test-placeholder"
      />
    );
    const input = screen.getByRole('textbox');

    expect(input).toHaveValue(value);
    input && expect(input).toBeEnabled();
    const updatedValue = 'Updated Value';
    fireEvent.change(input, { target: { value: updatedValue } });
    expect(input).toHaveValue(updatedValue);
  });
});
