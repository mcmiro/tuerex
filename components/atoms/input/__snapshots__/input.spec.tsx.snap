// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test input component Should pass when name, variant, size and placeholder is additional 1`] = `
<input
  autocomplete="off"
  class="w-full outline-none text-md font-normal bg-transparent"
  name="test"
  placeholder="test-placeholder"
  value=""
/>
`;
